client/cmd/hack/main.go

```diff
@@ -0,0 +1,313 @@
+	go func() {
+		defer wg.Done()
+		rpcserver.SetLogger(log)
+		rpcCfg := &rpcserver.Config{
+			Core: clientCore,
+			Addr: cfg.RPCAddr,
+			User: cfg.RPCUser,
+			Pass: cfg.RPCPass,
+			Cert: cfg.RPCCert,
+			Key:  cfg.RPCKey}
+		rpcSrv, err := rpcserver.New(rpcCfg)
+		if err != nil {
+			log.Errorf("Error starting rpc server: %v", err)
+			cancel()
+			return
+		}
+		rpcSrv.Run(ctx)
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		webSrv, err := webserver.New(clientCore, cfg.WebAddr, logMaker.Logger("WEBS"), true)
+		if err != nil {
+			log.Errorf("Error starting web server: %v", err)
+			cancel()
+			return
+		}
+		webSrv.Run(ctx)
+	}()
+
+	shutItDown := func() {
+		cancel()
+		wg.Wait()
+		coreRunner.Stop()
+		coreRunner.WaitForShutdown()
+	}
+
+	// Do stuff.
+	user := clientCore.User()
+	if user == nil || !user.Initialized {
+		err = clientCore.InitializeClient(clientPW)
+		if err != nil {
+			shutItDown()
+			return fmt.Errorf("InitializeClient error: %v", err)
+		}
+	}
+
+	// !user.Authed
+	negotiations, err := clientCore.Login(clientPW)
+	if err != nil {
+		shutItDown()
+		return fmt.Errorf("Login error: %v", err)
+	}
+
+	log.Infof("Logged in, authenticated with %d DEXs.", len(negotiations))
+
+	var regFee uint64
+	if len(negotiations) == 0 {
+		regFee, err = clientCore.PreRegister(dexHostPort)
+		if err != nil {
+			shutItDown()
+			return fmt.Errorf("Login error: %v", err)
+		}
+		log.Infof("Connected to dex %s, requiring fee %d", dexHostPort, regFee)
+	}
+
+	enumWallets := func() {
+		// clientCore.SupportedAssets first
+		// also clientCore.Balance
+		wallets := clientCore.Wallets()
+		log.Infof("Found %d wallet(s).", len(wallets))
+		for _, w := range wallets {
+			log.Infof("-Wallet %s (%d): running=%v, open=%v", w.Symbol, w.AssetID, w.Running, w.Open)
+		}
+	}
+	enumWallets()
+
+	dcrID, _ := dex.BipSymbolID("dcr")
+	if len(negotiations) == 0 {
+		dcrwalletConf = cleanAndExpandPath(dcrwalletConf)
+		var created bool
+	checkYoSelf:
+		if state := clientCore.WalletState(dcrID); state == nil {
+			if created {
+				shutItDown()
+				return fmt.Errorf("we already tried creating the DCR wallet!")
+			}
+			log.Infof("No DCR wallet found. Creating...")
+			// no DCR wallet
+			err = clientCore.CreateWallet(clientPW, walletPW, &core.WalletForm{
+				AssetID: dcrID,
+				Account: "default",
+				INIPath: dcrwalletConf,
+			})
+			if err != nil {
+				shutItDown()
+				return fmt.Errorf("CreateWallet error: %v", err)
+			}
+			created = true
+			err = clientCore.OpenWallet(dcrID, clientPW)
+			if err != nil {
+				shutItDown()
+				return fmt.Errorf("OpenWallet error: %v", err)
+			}
+			goto checkYoSelf
+		} else if !state.Running || !state.Open {
+			err = clientCore.OpenWallet(dcrID, clientPW)
+			if err != nil {
+				shutItDown()
+				return fmt.Errorf("OpenWallet error: %v", err)
+			}
+			goto checkYoSelf
+			// shutItDown()
+			// return fmt.Errorf("start the dcrwallet using %s or edit the wallet config", dcrwalletConf)
```

Comment:
Can no longer open an existing wallet, but it worked prior to rebase.  It still finds the dcr wallet and does not try to create a new one, but here OpenWallet says:

```
OpenWallet: decryption error: aead.Open: chacha20poly1305: message authentication failed
```

BTW, this turned into an infinite loop if there are no errors but the wallet never opens.  Temporary while I figure this outj